src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -28,6 +29,7 @@ internal partial class LanguageParser : SyntaxParser
         private int _recursionDepth;
         private TerminatorState _termState; // Resettable
         private bool _isInTry; // Resettable
+        private bool _checkedSimpleProgramsFeatureAvailability; // Resettable
```

Comment:
> Is this just to prevent reporting an error for each statement?

Yes.

---
In reply to: [382803207](https://github.com/dotnet/roslyn/pull/41706#discussion_r382803207) [](ancestors = 382803207)